viewing_party/party.py

```diff
@@ -1,23 +1,197 @@
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    sum_avg_rating = 0.0
 
+    if len(user_data["watched"]) == 0:
+        return sum_avg_rating
+    else:
+        for movie in user_data["watched"]:
+            sum_avg_rating += movie["rating"]
+            avg_rating = sum_avg_rating / len(user_data["watched"])
+        return avg_rating
+    
+def get_most_watched_genre(user_data):
+    freq = {}
+    highest_count = 0
+
+    if len(user_data["watched"]) == 0:
+        return None
+
+    for movie in user_data["watched"]:
+        if movie["genre"] in freq:
+            freq[movie["genre"]] += 1
+        else: 
+            freq[movie["genre"]]= 1
+
+    for genre, count in freq.items():
+        if count > highest_count:
+            highest_count = count
+            highest_genre = genre
+    return highest_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+# 1. Create a function named `get_unique_watched`. This function should...
+# - take one parameter: `user_data`
+#   - the value of `user_data` will be a dictionary with a `"watched"` list of movie dictionaries, and a `"friends"`
+#     - This represents that the user has a list of watched movies and a list of friends
+#     - The value of `"friends"` is a list
+#     - Each item in `"friends"` is a dictionary. This dictionary has a key `"watched"`, which has a list of movie dictionaries.
+#     - Each movie dictionary has a `"title"`.
+# - Consider the movies that the user has watched, and consider the movies that their friends have watched. 
+#   Determine which movies the user has watched, but none of their friends have watched.
+# - Return a list of dictionaries, that represents a list of movies
+
+# 2. Create a function named `get_friends_unique_watched`. This function should...
+# - take one parameter: `user_data`
+#   - the value of `user_data` will be a dictionary with a `"watched"` list of movie dictionaries, and a `"friends"`
+#     - This represents that the user has a list of watched movies and a list of friends
+#     - The value of `"friends"` is a list
+#     - Each item in `"friends"` is a dictionary. This dictionary has a key `"watched"`, which has a list of movie dictionaries.
+#     - Each movie dictionary has a `"title"`.
+# - Consider the movies that the user has watched, and consider the movies that their friends have watched. 
+# Determine which movies at least one of the user's friends have watched, but the user has not watched.
+# - Return a list of dictionaries, that represents a list of movies
+
+def get_unique_watched(user_data):
+    unique_movie = []
+    friends_watched = user_data["friends"]
+    friend_watch_list= []
+    user_watched = user_data["watched"]
+
+    for friends in friends_watched:
+        for movie in friends["watched"]:
+            if movie["title"] not in friend_watch_list:

```

Comment:
These nested lines will give you a time complexity of `0(n^3)`, one suggestion that i think would help you get the time complexity down for this maybe implementing a helper function that allows for you to handle maybe getting all of the movies in inside of the friends watch list. That way you could just have one loop that sees if the user's movies are in that list. This helper function can be used in `get_friends_unqiue_watched` as well and circumvent the `O(n^3)` code block as well!